.github/bin/concepts/update.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env sh
+
+# Update the concepts summary
+dotnet run -p .github/bin/concepts
+
+# Format the documents
+npx prettier@2.0.4 --write reference/README.md reference/references.json
+
+# Add the updated concept summary files
+git add reference/README.md
+git add reference/references.json
+
+# Check if there is nothing to commit (i.e. no changes to the concepts summary)
+if [ -z "$(git status --porcelain)" ]; then
+    echo "No changes to the concepts summary"
+    exit 0
+fi
+
+# Checkout to new branch
+BRANCH="conceptssummary-$(date +%Y%m%d%H%M%S)"
+git checkout -b "$BRANCH"
+
+# Setup the git user (required to commit anything)
+git config --global user.email "github-actions[bot]@users.noreply.github.com"
+git config --global user.name "github-actions[bot]"
+
+# Commit and push the changes
+git commit -m "[CI] Update concepts summary"
```

Comment:
Unrelated to this PR: Could we add a `[BOT]` tag or sth like that to easily filter auto-generated commits? `[CI]` is often used for changes to CI.

Reply:
Good idea. I'll update in a follow-up PR.